stellargraph/datasets/dataset_loader.py

```diff
@@ -0,0 +1,218 @@
+
+    def __init__(self) -> None:
+        # basic check since this is effectively an abstract base class, and derived classes should have set name
+        if not self.name:
+            raise ValueError(
+                f"{self.__class__.__name__} can't be instantiated directly, please use a derived class"
+            )
+
+    @property
+    def base_directory(self) -> str:
+        """str: The full path of the directory containing this dataset."""
+        return os.path.join(self._all_datasets_directory(), self.directory_name)
+
+    @property
+    def data_directory(self) -> str:
+        """str: The full path of the directory containing the data content files for this dataset."""
+        if self.data_subdirectory_name is None:
+            return self.base_directory
+        else:
+            return os.path.join(self.base_directory, self.data_subdirectory_name)
+
+    def _create_base_directory(self) -> None:
+        data_dir = self.base_directory
+        if not os.path.exists(data_dir):
+            os.makedirs(data_dir)
+
+    @staticmethod
+    def _all_datasets_directory() -> str:
+        """Return the path of the base directory which contains subdirectories for each dataset."""
+        return os.path.expanduser(
+            os.getenv(
+                "STELLARGRAPH_DATASETS_PATH", os.path.join("~", "stellargraph-datasets")
+            )
+        )
+
+    def _resolve_path(self, filename: str) -> str:
+        """Convert dataset relative file names to their full path on filesystem"""
+        return os.path.join(self.base_directory, filename)
+
+    def _is_downloaded(self) -> bool:
+        """Returns true if the expected files for the dataset are present"""
+        return all(
+            os.path.isfile(self._resolve_path(file)) for file in self.expected_files
+        )
+
+    def _verify_files_downloaded(self) -> None:
+        """
+        Raises:
+            FileNotFoundError: If any files within dataset are missing.
+        """
+        missing_files = ", ".join(
+            [
+                file
+                for file in self.expected_files
+                if not os.path.isfile(self._resolve_path(file))
+            ]
+        )
+        if missing_files:
+            raise FileNotFoundError(
+                f"{self.name} dataset failed to download file(s): {missing_files} to {self.data_directory}"
+            )
+
+    def _delete_existing_files(self) -> None:
+        """ Delete the files for this dataset if they already exist """
+        for file in self.expected_files:
+            try:
+                os.remove(self._resolve_path(file))
+            except OSError:
+                pass
+
+    def download(self, ignore_cache: Optional[bool] = False) -> None:
+        """
+        Download the dataset (if not already downloaded)
+
+        Args:
+            ignore_cache bool, optional (default=False): Ignore a cached dataset and force a re-download.
```

Comment:
I suspect this is meant to be something like:

```suggestion
            ignore_cache (bool, optional, default=False): Ignore a cached dataset and force a re-download.
```

but the RTD build isn't working to check.

In addition, I believe the default value should be listed in the function signature:

```suggestion
            ignore_cache (bool, optional): Ignore a cached dataset and force a re-download.
```